### ISSUE TITLE
PHP8.2. UriHelper.php:36. Function utf8_decode() is deprecated.

### ISSUE THREAD

**Issue by ReLater** [joomla-framework/uri#34]
### Steps to reproduce the issue
- Joomla 4.2.6
- PHP 8.2.0
- In a scenario like

```
use Joomla\CMS\Uri\Uri;
...
$this->redirectUrl = Uri::getInstance()->toString();
...
```

a deprecated message occurs:

> Function utf8_decode() is deprecated. UriHelper.php:36 (method parse_url()).

**Comment by ReLater** [comment]
Related: https://github.com/joomla/joomla-cms/issues/38922

**Comment by laoneo** [comment]
If this one is accepted https://github.com/joomla/joomla-cms/pull/39583, then we can do the same fix here as well.

**Comment by laoneo** [comment]
Can be closed as we have a pr #35
